scripts/apply_style.py

```
@@ -249,70 +218,50 @@ def apply_style_to_file(source_path):
-        output('Applying style...')
-    else:
-        handler = diff_style_to_file
-        if args.names_only:
-            output('Files to reformat:')
-        else:
-            output('Checking style...')
-
-    # process all the files
-    success = False
+    if args.names_only:
+        output('Files to reformat:')
 
     processed_files = args.filename or project_sources(PROJECT_ROOT)
 
     with ThreadPoolExecutor(max_workers=args.jobs) as pool:
-        result = pool.map(handler, processed_files)
+        result = pool.map(apply_style_to_file, processed_files)
 
         if args.all:
             result = list(result)
 
-        success = all(result)
+        all(result)
 
-    if args.fix:
-        output('Applying style complete!')
-    else:
-        output('Checking style complete!')
 
-    if not success:
-        sys.exit(1)
+def format_python(args):
+    jobs_arg = ['-j {}'.format(args.jobs)]
 
+    autopep8_cmd = ['autopep8', '.', '--in-place', '--recursive',
+                    '--exclude', 'vendor'] + jobs_arg
 
-def format_python(args):
-    fix_or_diff_arg = ['--in-place' if args.fix else '--diff']
-    # Parallel jobs requires '--in-place' arg
-    jobs_arg = ['-j {}'.format(args.jobs)] if args.fix else []
+    subprocess.call(autopep8_cmd, cwd=PROJECT_ROOT)
 
-    autopep8_cmd = ['autopep8', '.', '--exit-code', '--recursive',
-                    '--exclude', 'vendor'] + fix_or_diff_arg + jobs_arg
 
-    exit_code = subprocess.call(autopep8_cmd, cwd=PROJECT_ROOT)
-    if exit_code != 0:
-        sys.exit(1)
+def get_diff():
+    return subprocess.check_output(['git', 'diff'], cwd=PROJECT_ROOT).decode().strip()
 
 
 def main():
     args = parse_commandline()
 
     # TODO(WK) Make multilanguage reformatting concurrent
+    output('Formatting C/C++ ...')
     format_cpp(args)
+    output('Formatting Python ...')
     format_python(args)
+    output('Done.')
+
+    if args.diff:
+        diff = get_diff()
```

Is there a non-zero exit on diff output here?

---

I'm not sure if you are asking about the script or `git diff` invoked in `get_diff()`.

* The exit code of the script is non-zero if we're in diff mode and the diff is non-empty, i.e. if the reformatting resulted in changes. This will let the script fail the Jenkins build when running CI. If we're not in diff mode, but just a developer wanting to bring their local copy into line with the style, then the exit code is 0 regardless of whether there have been changes made or not.

* `git diff` on the other hand, returns a zero exit code upon successful execution, regardless of whether it finds any changes or not.